test/testcafe/spec/v1/DeviceCodeActivate_spec.js

```diff
@@ -0,0 +1,182 @@
+  await deviceCodeActivatePage.navigateToPage();
+  requestLogger.clear();
+  return deviceCodeActivatePage;
+}
+
+test.requestHooks(requestLogger, legacyDeviceCodeSuccessMock)('should be able to complete device code activation flow on legacy SIW', async t => {
+  const deviceCodeActivatePageObject = await setup(t);
+  await rerenderWidget({
+    stateToken: '00-dummy-state-token', //start with 00 to render legacy sign in widget
+  });
+
+  await t.expect(deviceCodeActivatePageObject.getPageTitle()).eql('Activate your device');
+  await t.expect(deviceCodeActivatePageObject.getPageSubtitle()).eql('Follow the instructions on your device to get an activation code');
+  await t.expect(await deviceCodeActivatePageObject.getActivationCodeTextBoxLabel()).eql('Activation Code');
+  await t.expect(deviceCodeActivatePageObject.isActivateCodeTextBoxVisible()).eql(true);
+
+  await t.expect(requestLogger.contains(record => record.request.url.match(/api\/v1\/authn\/introspect/))).eql(true);
+
+  requestLogger.clear();
+
+  // submit user code
+  await deviceCodeActivatePageObject.setActivateCodeTextBoxValue('ABCDWXYZ');
+  await deviceCodeActivatePageObject.clickNextButton();
+
+  await t.expect(requestLogger.count(() => true)).eql(1);
+  const req = requestLogger.requests[0].request;
+  const reqBody = JSON.parse(req.body);
+  await t.expect(reqBody).eql({
+    userCode: 'ABCD-WXYZ',
+    stateToken: '00-dummy-state-token',
+  });
+  await t.expect(req.method).eql('post');
+  await t.expect(req.url).eql('http://localhost:3000/api/v1/authn/device/activate');
+
+  requestLogger.clear();
+
+  // identify with password
+  await deviceCodeActivatePageObject.fillLegacyUserNameField('Test Identifier');
+  await deviceCodeActivatePageObject.fillLegacyPasswordField('random password 123');
+  await deviceCodeActivatePageObject.clickNextButton();
+
+  await t.expect(requestLogger.count(() => true)).eql(1);
+  const reqIdentify = requestLogger.requests[0].request;
+  await t.expect(reqIdentify.method).eql('post');
+  await t.expect(reqIdentify.url).eql('http://localhost:3000/api/v1/authn');
+
+  // expect device activated screen
+  await t.expect(deviceCodeActivatePageObject.getPageTitle()).eql('Device activated');
+  await t.expect(deviceCodeActivatePageObject.getPageSubtitle()).eql('Follow the instructions on your device for next steps');
+  await t.expect(deviceCodeActivatePageObject.isTerminalSuccessIconPresent()).eql(true);
+  await t.expect(deviceCodeActivatePageObject.isBeaconTerminalPresent()).eql(false);
+  await t.expect(deviceCodeActivatePageObject.isTryAgainButtonPresent()).eql(false);
+});
+
+test.requestHooks(legacyDeviceCodeConsentDeniedMock)('should be able to get device not activated screen when consent is denied on legacy SIW', async t => {
+  const deviceCodeActivatePageObject = await setup(t);
+  await rerenderWidget({
+    stateToken: '00-dummy-state-token', //start with 00 to render legacy sign in widget
```

Comment:
nit:
ideally, we could think of creating a whole new suite for running v1 testcafe tests.
e.g. `test:testcafe-ci-v1` which create widgetrc with v1 token and run specs only from `specs/v1` etc.

It's not really related to your changes but if you're interested, certainly appreciate to tackle it in separate PR.